### PR TITLE
Update CiApp tags according the specs.

### PR DESCRIPTION
**Pull request by tonyredondo**
This PR updates the CIApp environment variables mapping according the specifications updates.

@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs [hunk 1]
```diff
@@ -53,17 +57,60 @@ static CIEnvironmentValues()
...
+            // **********
+
+            SourceRoot = ExpandPath(SourceRoot);
+            WorkspacePath = ExpandPath(WorkspacePath);
+
+            // **********
+            // Clean Refs
+            // **********
+
+            var regex = new Regex(@"^refs\/heads\/(.*)|refs\/tags\/(.*)|refs\/(.*)$", RegexOptions.Compiled);
```
Comment (kevingosse) [inline src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs:82]: The regex is invoked only a couple times, compiling it is probably more expensive than just evaluating it

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs [hunk 2]
```diff
@@ -107,19 +158,71 @@ public static void DecorateSpan(Span span)
...
-            span.SetTag(CommonTags.GitCommit, Commit);
-            span.SetTag(CommonTags.GitBranch, Branch);
-            span.SetTag(CommonTags.GitTag, Tag);
-
-            span.SetTag(CommonTags.BuildSourceRoot, SourceRoot);
+
+            if (!string.IsNullOrEmpty(PipelineId))
+            {
+                span.SetTag(CommonTags.CIPipelineId, PipelineId);
+            }
```
Comment (kevingosse) [inline src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs:165]: NIT: might be worth introducing a helper method 😛

**Review by kevingosse — APPROVED** [review]

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs (hunk 1 above)
Comment (tonyredondo) [reply src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs]: done in 3215416

File: src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs (hunk 2 above)
Comment (tonyredondo) [reply src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs]: done in 3215416

**Review by kevingosse — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**